patches/unique-db.diff

```diff
@@ -1,36 +0,0 @@
-Prevent state collisions
-
-Previously if you opened different workspaces that had the same filesystem path
-(for example if you have /home/coder on two different machines that are both
-accessed through the same host) they would conflict with each other.  This
-ensures that different browser paths will be unique (for example /workspace1 and
-/workspace2).
-
-The easiest way to test is to open files in the same workspace using both / and
-/vscode and make sure they are not interacting with each other.
-
-Index: code-server/lib/vscode/src/vs/workbench/services/storage/browser/storageService.ts
-===================================================================
---- code-server.orig/lib/vscode/src/vs/workbench/services/storage/browser/storageService.ts
-+++ code-server/lib/vscode/src/vs/workbench/services/storage/browser/storageService.ts
-@@ -17,6 +17,7 @@ import { AbstractStorageService, isProfi
- import { IUserDataProfile } from 'vs/platform/userDataProfile/common/userDataProfile';
- import { IAnyWorkspaceIdentifier } from 'vs/platform/workspace/common/workspace';
- import { IUserDataProfileService } from 'vs/workbench/services/userDataProfile/common/userDataProfile';
-+import { hash } from 'vs/base/common/hash';
- 
- export class BrowserStorageService extends AbstractStorageService {
- 
-@@ -67,7 +68,11 @@ export class BrowserStorageService exten
- 					return `global-${this.profileStorageProfile.id}`;
- 				}
- 			case StorageScope.WORKSPACE:
--				return this.payload.id;
-+				// Add a unique ID based on the current path for per-workspace databases.
-+				// This prevents workspaces on different machines that share the same domain
-+				// and file path from colliding (since it does not appear IndexedDB can be
-+				// scoped to a path) as long as they are hosted on different paths.
-+				return this.payload.id + '-' + hash(location.pathname.toString().replace(/\/$/, "")).toString(16);
```

Comment:
Oh, good point. I wasn't sure on that. Let me revert this.

Reply:
Hmm...thought this would be easier. So talking out loud for a second. They modified some of this logic here, I think: https://github.com/microsoft/vscode/commit/ab8a8ced8231af37b07257c2868dacb38baf2095#diff-3bbec17f0545b2ead2ce7e5ba84d0bb2807d1d5a4a4a00d1979570dbc127b912

Since they're still using `this.payload.id` for the workspace storage, I believe we still need our patch. I will update.